### PR TITLE
Backport of Bump Envoy Versions  into release/1.18.x

### PR DESCRIPTION
**Pull request by hc-github-team-consul-core**
## Backport

This PR is auto-generated from #21277 to be assessed for backporting due to the inclusion of the label backport/1.18.

:rotating_light:
>**Warning** automatic cherry-pick of commits failed. If the first commit failed,
you will see a blank no-op commit below. If at least one commit succeeded, you
will see the cherry-picked commits up to, _not including_, the commit where
the merge conflict occurred.

The person who merged in the original PR is:
@sarahalsmiller
This person should manually cherry-pick the original PR into a new backport PR,
and close this one when the manual backport PR is merged in.

> merge conflict error: POST https://api.github.com/repos/hashicorp/consul/merges: 409 Merge conflict []

The below text is copied from the body of the original PR.

---

### Description

Bump Envoy versions 
### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

---

<details>
<summary> Overview of commits </summary>

  - 44412be4a312f341f74b69234cc7b24610f0c6b4  - 3f26712527812790fa1402bf73d6538beb9ab7c2  - 16b4fa130127c06b521ce01aa2ddf7f946710e9b 

</details>

### CODE REVIEW

**Review by github-team-consul-core-pr-approver — APPROVED** [review]
Auto approved Consul Bot automated PR

File: test/integration/connect/envoy/run-tests.sh [hunk 1]
```diff
@@ -15,7 +15,7 @@ DEBUG=${DEBUG:-}
 XDS_TARGET=${XDS_TARGET:-server}
 
 # ENVOY_VERSION to run each test against
-ENVOY_VERSION=${ENVOY_VERSION:-"1.27.0"}
+ENVOY_VERSION=${ENVOY_VERSION:-"1.28.4"}
```
Comment (nathancoleman) [inline test/integration/connect/envoy/run-tests.sh:18]: It looks like the jump here is due to this update being missed in the past. Is that correct?

File: .github/workflows/test-integrations.yml [hunk 2]
```diff
@@ -270,7 +270,11 @@ jobs:
           # this is further going to multiplied in envoy-integration tests by the
           # other dimensions in the matrix.  Currently TOTAL_RUNNERS would be
           # multiplied by 2 based on these values:
+<<<<<<< HEAD
```
Comment (nathancoleman) [inline .github/workflows/test-integrations.yml]: Some merge conflicts in this file that need to be resolved

**Review by nathancoleman — CHANGES_REQUESTED** [review]

File: .github/workflows/test-integrations.yml [hunk 3]
```diff
@@ -305,7 +305,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        envoy-version: ["1.28.3"]
+        envoy-version: ["1.29.5"]
```
Comment (nathancoleman) [inline .github/workflows/test-integrations.yml:308]: Should this be `1.28.4`? It [looks like](https://developer.hashicorp.com/consul/docs/connect/proxies/envoy#standard-releases) Consul 1.18 doesn't support Envoy `1.29.5`

File: test/integration/connect/envoy/run-tests.sh (hunk 1 above)
Comment (sarahalsmiller) [reply test/integration/connect/envoy/run-tests.sh:18]: Yes I am assuming so

File: .github/workflows/test-integrations.yml [hunk 4]
```diff
@@ -270,7 +270,7 @@ jobs:
           # this is further going to multiplied in envoy-integration tests by the
           # other dimensions in the matrix.  Currently TOTAL_RUNNERS would be
           # multiplied by 2 based on these values:
-          # envoy-version: ["1.28.3"]
+          # envoy-version: ["1.29.5"]
```
Comment (sarahalsmiller) [inline .github/workflows/test-integrations.yml:273]:
```suggestion
          # envoy-version: ["1.28.4"]
```

**Review by nathancoleman — APPROVED** [review]

**Pull request merged**